### ISSUE TITLE
How to handle a stream response for a job request?

### ISSUE THREAD

**Issue by thatguyjono94** [maclof/kubernetes-client#58]
Looking at: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#-strong-read-operations-strong--43

There is a `watch` flag that says `Watch for changes to the described resources and return them as a stream of add, update, and remove notifications. Specify resourceVersion.`

I need functionality so that I will get the response when the job has finished. Is it possible to achieve this with the API here? Any code examples would be great.

**Comment by thatguyjono94** [comment]
Any follow up on this?

**Comment by maclof** [comment]
@thatguyjono94 sorry for long delay, I will be looking at this asap

**Comment by maclof** [comment]
@thatguyjono94 As per the following commit this is now partially supported: https://github.com/maclof/kubernetes-client/commit/d67bf78fea510253e91f4a9407ce339e3943adfd

I have some more testing that I'd like to do, but you're able to watch any object and a closure you provide will be called for each response (ADDED, UPDATED, DELETED, etc)

Example below, the timeoutSeconds parameter is optional.

```php
$client->cronJobs()->watch($cronJob, function ($type, $object) {
	var_dump('cronJob', $type, $object['metadata']['name']);
}, [
	'timeoutSeconds' => 120,
]);
```

**Comment by maclof** [comment]
This is included in release 0.18.0
